src/firecracker/src/main.rs

```diff
@@ -40,6 +40,13 @@ fn main() {
         process::exit(i32::from(vmm::FC_EXIT_CODE_GENERIC_ERROR));
     }
 
+    // Make stdout non blocking.
+    let flags = unsafe { libc::fcntl(libc::STDOUT_FILENO, libc::F_GETFL, 0) };
+    let rc = unsafe { libc::fcntl(libc::STDOUT_FILENO, libc::F_SETFL, flags | libc::O_NONBLOCK) };
+    if rc < 0 {
+        error!("Could not set Firecracker stdout to non-blocking.");
+    }
```

Comment:
Please move this code to `src/vmm/builder.rs` next to calling `builder::setup_serial_device()` to keep logic contained.

`setup_serial_device()` can take any `out: Box<dyn io::Write + Send>` as output, we set the output as being `STDOUT_FILENO` in builder so that's where we should also make `STDOUT_FILENO` `O_NONBLOCK`.

Reply:
Done.